src/v/container/fragmented_vector.h

```diff
@@ -480,6 +480,15 @@ class fragmented_vector {
      */
     void erase_to_end(const_iterator begin) { pop_back_n(cend() - begin); }
 
+    template<typename Range>
+    requires std::ranges::range<Range>
```

Comment:
`std::ranges::sized_range` to avoid repeated scanning for the size? And if we need it for these one day another override should be added, probably without `reserve` call.